cli/js/errors.ts

```diff
@@ -45,39 +44,13 @@ export enum ErrorKind {
   Other = 17,
   UnexpectedEof = 18,
   BadResource = 19,
-  CommandFailed = 20,
-  EmptyHost = 21,
-  IdnaError = 22,
-  InvalidPort = 23,
-  InvalidIpv4Address = 24,
-  InvalidIpv6Address = 25,
-  InvalidDomainCharacter = 26,
-  RelativeUrlWithoutBase = 27,
-  RelativeUrlWithCannotBeABaseBase = 28,
-  SetHostOnCannotBeABaseUrl = 29,
-  Overflow = 30,
-  HttpUser = 31,
-  HttpClosed = 32,
-  HttpCanceled = 33,
-  HttpParse = 34,
-  HttpOther = 35,
-  TooLarge = 36,
-  InvalidUri = 37,
-  InvalidSeekMode = 38,
-  OpNotAvailable = 39,
-  WorkerInitFailed = 40,
-  UnixError = 41,
-  NoAsyncSupport = 42,
-  NoSyncSupport = 43,
-  ImportMapError = 44,
-  InvalidPath = 45,
-  ImportPrefixMissing = 46,
-  UnsupportedFetchScheme = 47,
-  TooManyRedirects = 48,
-  Diagnostic = 49,
-  JSError = 50,
-  TypeError = 51,
-
-  /** TODO this is a DomException type, and should be moved out of here when possible */
-  DataCloneError = 52
+  UrlParse = 20,
+  Http = 21,
+  TooLarge = 22,
+  InvalidSeekMode = 23,
+  UnixError = 24,
```

Comment:
I don't think we should fold all OS errors under 'UnixError'. It's unpractical when e.g. renaming a file fails with some generic error because you might want to handle it based on the specific reason for the failure, e.g. `rmdir` could fail because the directory was not found (so you're good) or because it's a file or because the dir is not empty.

Reply:
Agree, it seems we've been using `UnixError` for quite some time. We should decide which of error code provided by Rust `Error::Sys` we want to support and list them collapsing the rust under `ErrorKind.SysError`